src/o365/spo/commands/tenant/tenant-settings-set.ts

```diff
@@ -0,0 +1,861 @@
+
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((resp: string): request.RequestPromise => {
+        accessToken = resp;
+
+        if (this.debug) {
+          cmd.log(`Retrieved access token ${accessToken}.`);
+        }
+
+        return this.getRequestDigest(cmd, this.debug);
+      })
+      .then((contextResponse: ContextInfo): request.RequestPromise => {
+
+        if (this.debug) {
+          cmd.log('Retrieved ContextInfo...');
+          cmd.log(contextResponse);
+          cmd.log('');
+        }
+
+        formDigestValue = contextResponse.FormDigestValue;
+
+        const requestOptions: any = {
+          url: `${auth.site.url}/_vti_bin/client.svc/ProcessQuery`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${auth.service.accessToken}`,
+            'X-RequestDigest': formDigestValue
+          }),
+          body: `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><ObjectPath Id="4" ObjectPathId="3" /><Query Id="5" ObjectPathId="3"><Query SelectAllProperties="true"><Properties /></Query></Query></Actions><ObjectPaths><Constructor Id="3" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}" /></ObjectPaths></Request>`
+        };
+
+        if (this.debug) {
+          cmd.log('Retrieved ContextInfo...');
+          cmd.log(contextResponse);
+          cmd.log('');
+        }
+
+        return request.post(requestOptions);
+      })
+      .then((res: any): request.RequestPromise | Promise<any> => {
+
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(res);
+          cmd.log('');
+        }
+
+        const json: ClientSvcResponse = JSON.parse(res);
+        const response: ClientSvcResponseContents = json[0];
+        if (response.ErrorInfo) {
+          const error: string = response.ErrorInfo.ErrorMessage;
+          return new Promise<any>((resolve, reject) => { reject(error) });
+        }
+
+        // map the args.options to XML Properties
+        let propsXml: string = '';
+        let id = 42; // geek's humor
+        for (let optionKey of Object.keys(args.options)) {
+
+          if (this.isExcludedOption(optionKey)) {
+            continue;
+          }
+
+          let optionValue = args.options[optionKey];
+          if (this.getAllEnumOptions().indexOf(optionKey) > -1) {
+            // map enum values to int
+            optionValue = this.mapEnumToInt(optionKey, args.options[optionKey]);
+          }
+
+          if (['AllowedDomainListForSyncClient', 'DisabledWebPartIds'].indexOf(optionKey) > -1) {
+
+            // the XML has to be represented as array of guids
+            let valuesXml = '';
+            optionValue.split(',').forEach((value: string) => {
+              valuesXml += `<Object Type="Guid">{${value}}</Object>`;
```

Comment:
Here, we should escape any XML that could be included in the `value` variable using `Utils.escapeXml()`

Reply:
Sorry, I keep forgetting about that.

Reply:
No worries. I miss it sometimes as well. This is why it's helpful to have another pair of eyes go through the code 🙂